msal/src/main/java/com/microsoft/identity/client/PublicClientApplication.java

```diff
@@ -1618,6 +1624,38 @@ public void onError(MsalException exception) {
         }
     }
 
+    public void deviceCodeFlow(@Nullable String[] scopes, @NonNull final DeviceCodeFlowCallback callback) {
+        // Create a DeviceCodeFlowCommandParameters object that takes in the desired scopes and the callback object
+        // Use CommandParametersAdapter
+        final DeviceCodeFlowCommandParameters commandParameters = CommandParametersAdapter
+                    .createDeviceCodeFlowCommandParameters(
+                            mPublicClientConfiguration,
+                            mPublicClientConfiguration.getOAuth2TokenCache(),
+                            scopes);
+
+        // Create a CommandCallback object from the DeviceCodeFlowCallback object
+        final DeviceCodeFlowCommandCallback deviceCodeFlowCommandCallback = getDeviceCodeFlowCommandCallback(callback);
+
+        // Create a DeviceCodeFlowCommand object
+        // Pass the command parameters, default controller, and command callback
+        // Telemetry with DEVICE_CODE_FLOW_CALLBACK
+        final DeviceCodeFlowCommand deviceCodeFlowCommand = new DeviceCodeFlowCommand(
+                commandParameters,
+                new LocalMSALController(),
+                deviceCodeFlowCommandCallback,
+                PublicApiId.DEVICE_CODE_FLOW_WITH_CALLBACK
+        );
+
+        // Run the command we created above in a separate thread to allow running HTTP Requests
+        Thread thread = new Thread(new Runnable(){
+            @Override
+            public void run(){
+                CommandDispatcher.submitSilent(deviceCodeFlowCommand);
+            }
+        });
+        thread.start();
```

Comment:
Can you remind me why we need a new thread here? CommandDispatcher should be spinning up its own threads any way so I believe there shouldn't be any HTTP request that is made on the main thread.

Reply:
You may be right about this. This may just be a remnant from the Proof of Concept. I'll test this without the thread in a bit.

Reply:
`CommandDispatcher.submitSilent()` documentation states that it uses a silent thread pool, so I will remove the new thread in PCA.